src/test/java/com/revature/services/impl/AddressServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+package com.revature.services.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.revature.models.Address;
+import com.revature.models.AddressDTO;
+import com.revature.repositories.AddressRepository;
+
+@RunWith(SpringRunner.class)
+public class AddressServiceImplTest {
+	
+	@InjectMocks
+	private AddressServiceImpl asi;
+	
+	@Mock
+	private AddressRepository ar;
+	
+	@Test
+	public void testFindAll() {
+		List<Address> expected = new ArrayList<>();
+		expected.add(new Address());
+		expected.add(new Address());
+		when(ar.findAll()).thenReturn(expected);
+		List<Address> actual = asi.findAll();
+		
+		verify(ar).findAll();
+		assertEquals(2, actual.size());	
+	}
+	
+	@Test
+	public void testFindAllDTO() {
+		List<Address> expected = new ArrayList<>();
+		expected.add(new Address());
+		expected.add(new Address());
+		when(ar.findAll()).thenReturn(expected);
+		List<AddressDTO> actual = asi.findAllDTO();
+		
+		verify(ar).findAll();
+		assertEquals(2, actual.size());
+		
+	}
+	
+	@Test
+	public void testAddAddress() {
+		Address expected = new Address();
+		when(ar.save(expected)).thenReturn(expected);
+		Address actual = asi.addAddress(expected);
+		
+		verify(ar).save(expected);
```

Comment:
I believe the verify(ar).someMethod() can actually be appended with a times(x) method to verify that the dependency invoked the correct method a certain number of times. Might be an interesting addition to the tests.

Reply:
verify(ar) is the same as verify(ar, times(1)).
The dependencies are expected to only be invoked once as well.

Reply:
Ah ok, that's fine then.